webapp/login.go

```diff
@@ -0,0 +1,323 @@
+// Copyright 2019 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package webapp
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/gob"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/gorilla/securecookie"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"golang.org/x/oauth2"
+	ghOAuth "golang.org/x/oauth2/github"
+	"google.golang.org/appengine"
+)
+
+func init() {
+	// Need RegisterName for securecookie encoding - for local packages, Register appends main[0-9]{5}.User
```

Comment:
I'm confused by the latter half of this comment, which seems like an implementation detail of `gob`. Perhaps we can just leave it out. And make it clear that we need to register all custom types stored in `securecookie`.